src/cpp/session/SessionHttpMethods.cpp

```diff
@@ -844,6 +940,49 @@ void registerGwtHandlers()
                                                   initJs);
 }
 
+namespace {
+
+/**
+ * When a session debug logging preference is enabled, either create a new FileLog or update an existing one
+ * to have DEBUG so these messages show up. This prevents the user from needing to configure logging.conf
+ * for the session just to turn on debug logging.
+ */
+void initSessionDebugLog()
+{
+   if (s_sessionDebugLogCreated)
+      return;
+   s_sessionDebugLogCreated = true;
+
+   // TODO: should we have a preference for the path to this log file?
```

Comment:
Generally we avoid committing TODOs since it makes it hard to use them to check for remaining work in a PR. 

Since this is a dev debug mechanism I don't think it needs to be configurable; if someone really wants to put the logs somewhere else they can always use the XDG vars.